input/common/execread/execread.go

```diff
@@ -66,60 +79,83 @@ func (s *Session) Start(ctx context.Context, dst [][]input.Sample, kickChan chan
 		f64:   !s.f32mode,
 	}
 
-	// Allocate 4 times the buffer. We should ensure that we can read some of
-	// the overflow.
-	raw := make([]byte, bufsz)
-
-	if err := cmd.Start(); err != nil {
-		return errors.Wrap(err, "failed to start ffmpeg")
+	bufsz := s.samples
+	if !s.f32mode {
+		bufsz *= 2
 	}
 
-	for {
-		reader.reset(raw)
+	raw := make([]byte, bufsz*4)
 
-		mu.Lock()
-		for n := 0; n < s.samples; n++ {
-			dst[n%framesz][n/framesz] = reader.next()
-		}
-		mu.Unlock()
+	sampleDuration := time.Duration(
+		float64(s.cfg.SampleSize) / s.cfg.SampleRate * float64(time.Second))
 
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-			// default:
-		case kickChan <- true:
+	for {
+		// Set us a read deadline. If the deadline is reached, we'll write zeros
+		// to the buffer.
+		var writeZeros bool
+		if err := of.SetReadDeadline(time.Now().Add(sampleDuration)); err != nil {
+			return errors.Wrap(err, "failed to set read deadline")
 		}
 
 		_, err := io.ReadFull(o, raw)
```

Comment:
Currently this can read double the data, and causes the timing to be off.

Please change to `_, err := io.ReadAtLeast(o, raw, bufsz)`